apps/src/templates/teacherHomepage/AnnouncementsCarousel.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import FontAwesome from '../FontAwesome';
+import color from "../../util/color";
+import _ from 'lodash';
+
+const styles = {
+  box: {
+    width: 940,
+    backgroundColor: color.white
+  },
+  arrowIcon: {
+    fontSize: 40,
+    lineHeight: "40px",
+    color: color.white,
+    marginRight: 15,
+    marginLeft: 15
+  },
+  arrowBox: {
+    float: "right",
+    marginTop: 12,
+    marginRight: 25,
+    height: 40,
+    width: 90,
+    borderColor: color.white,
+    borderWidth: 1,
+    borderRadius: 3,
+    borderStyle: "solid"
+  },
+  arrowBoxRight: {
+    marginTop: 25,
+    marginRight: 25
+  },
+};
+
+const AnnouncementsCarousel = React.createClass({
+  propTypes: {
+    children: React.PropTypes.oneOfType([
+      React.PropTypes.node,
+      React.PropTypes.arrayOf(React.PropTypes.node)
+    ])
+  },
+
+  getInitialState() {
+    const announcements = this.props.children;
+    return { previous: announcements.length-1, current: 0, next: 1  };
```

Comment:
Also note that you should be able to do this pretty easily with modulo, i.e:
```
nextIndex = (currentIndex + delta + announcements.length) % announcements.length
```
where delta is -1 for previous and +1 for next


The reason for the extra `announcements.length` before the modulo, is that JS is kind of weird in that `-1 % 3 === -1` (Rather than 2).